File: test/integration/auth_test.py
import time
import os

import helpers
import uuid
import requests

class TestAuth(helpers.IntegrationTest):

    def __init__(self, *args, **kwargs):
        super(TestAuth, self).__init__(*args, **kwargs)
        self.username = "user-%s" % uuid.uuid4().hex
        self.password = self.username
        self.email = "%s@test.tld" % self.username

    def test_auth(self):
        print("test_auth")
        if os.name == 'nt':
            return
        self.auth_signup()
        self.auth_logout()
        self.auth_login()

    def test_helpers(self):
        print("test_helpers")
        if os.name == 'nt':
            return
        self.auth_logout()
        self.login_as_persistent_user()
        self.auth_logout()
        self.create_user_and_login()

    def auth_signup(self):
        print("auth_signup")
        self.spawn("auth signup")
        self.expect("username:")
        self.send(self.username)
        self.expect("password:")
        self.send(self.password)
        self.expect("again:")
        self.send(self.password)
        self.expect("name:")
        self.send(self.username)
        self.expect("email:")
        self.send(self.email)
        self.expect("account has been registered")
        self.wait()

    def auth_logout(self):
        print("auth_logout")
        self.spawn("auth logout")
        self.expect("You have been logged out")
        self.wait()

    def auth_login(self):
        print("auth_login")
        self.spawn("auth")
        self.expect("username:")
        self.send(self.username)
        self.expect("password:")
        self.send(self.password)
        self.expect("succesfully authenticated")
        self.wait()

        # still logged in?
        self.spawn("auth")
        self.expect("You are logged in")
        self.wait()

    def auth_login_with_flags(self):
        print("auth_login_with_flags")
        self.spawn("auth --username %s --password %s" % (self.username, self.password))
        self.expect("succesfully authenticated")
        self.wait()

        # still logged in?
        self.spawn("auth")
        self.expect("You are logged in")
        self.wait()

    def login_as_persistent_user(self):
        print("login_as_persistent_user")
        self.username = "cli-integration-tests"
        self.password = "test-cli-integration"
        self.auth_login_with_flags()

    def create_user_and_login(self):
        print("create_user_and_login")
        self.username, self.password = self.create_user()
        self.auth_login_with_flags()

    def create_user(self):
        print("create_user")
        username = "user-%s" % uuid.uuid4().hex
        data = {
            "name": username,
            "username": username,
            "password": username,
            "email": "%s@test.tld" % username,
            "EULAAccepted": True
        }
        r = requests.post("https://platform.activestate.com/api/v1/users", json=data)
        if r.status_code != 200:
            self.fail("Create user failed: code: %d, response: %s" % (r.status_code, r.content))
        return data["username"], data["password"]

if __name__ == '__main__':
    helpers.Run()

File: test/integration/auth_test.go
package integration_test

import (
	"fmt"
	"testing"

	"github.com/google/uuid"
	"github.com/stretchr/testify/suite"

	"github.com/ActiveState/cli/test/integration/expect"
)

var uid uuid.UUID

func init() {
	// Because of our parallel test running tweak normal suite operations are limited and we have to set up shared resources
	// at a higher level. We can better address this in the long run but for now this'll have to do.
	var err error
	uid, err = uuid.NewRandom()
	if err != nil {
		panic(fmt.Sprintf("Could not generate uuid, error: %v", err))
	}
}

type AuthTestSuite struct {
	Suite
	username string
	password string
	email    string
}

func (suite *AuthTestSuite) SetupTest() {
	suite.Suite.SetupTest()

	suite.username = fmt.Sprintf("user-%s", uid.String()[0:8])
	suite.password = suite.username
	suite.email = fmt.Sprintf("%s@test.tld", suite.username)
}

func (suite *AuthTestSuite) TestAuth() {
	suite.Signup()
	suite.Logout()
	suite.Login()
}

func (suite *AuthTestSuite) Signup() {
	suite.Spawn("auth", "signup")
	defer suite.Stop()

	suite.Expect("username:")
	suite.Send(suite.username)
	suite.Expect("password:")
	suite.Send(suite.password)
	suite.Expect("again:")
	suite.Send(suite.password)
	suite.Expect("name:")
	suite.Send(suite.username)
	suite.Expect("email:")
	suite.Send(suite.email)
	suite.Expect("account has been registered")
	suite.Wait()
}

func (suite *AuthTestSuite) Logout() {
	suite.Spawn("auth", "logout")
	defer suite.Stop()

	suite.Expect("You have been logged out")
	suite.Wait()
}

func (suite *AuthTestSuite) Login() {
	suite.Spawn("auth")
	suite.Expect("username:")
	suite.Send(suite.username)
	suite.Expect("password:")
	suite.Send(suite.password)
	suite.Expect("succesfully authenticated")
	suite.Wait()

	// still logged in?
	suite.Spawn("auth")
	suite.Expect("You are logged in")
	suite.Wait()
}

func TestAuthTestSuite(t *testing.T) {
	_ = suite.Run // vscode won't show test helpers unless I use this .. -.-

	//suite.Run(t, new(AuthTestSuite))
	expect.RunParallel(t, new(AuthTestSuite))
}
